Core/src/Visualization/GeometryView3D.cpp

```diff
@@ -34,32 +34,38 @@
 
 #include <algorithm>
 #include <cmath>
+#include <filesystem>
 #include <memory>
 #include <ostream>
 #include <utility>
 #include <vector>
 
-#include <limits.h>
-#include <unistd.h>
-
 namespace {
 
-std::string joinPaths(const std::string& a, const std::string& b) {
-  if (b.substr(0, 1) == "/" || a.empty()) {
-    return b;
+/// @brief Combines a base directory path with a relative file path to create an
+/// absolute path.
+/// - If the relative file path is already absolute, baseDir will be ignored.
+/// - If the baseDir is empty (which should not happen), we set it as the
+///   current working directory.
+///
+/// @param baseDir The base directory path.
+/// @param relativeFilePath The relative file path to be combined with the base directory.
+///
+/// @return A string containing the absolute path created by combining baseDir and relativeFilePath.
+std::string makeAbsPath(const std::string& baseDir,
+                        const std::string& relativeFilePath) {
+  std::filesystem::path pathBaseDir(baseDir);
+  const std::filesystem::path pathRelativeFilePath(relativeFilePath);
+
+  if (pathRelativeFilePath.is_absolute()) {
+    return pathRelativeFilePath.string();
   }
 
-  if (a.substr(a.size() - 1) == "/") {
-    return a.substr(a.size() - 1) + "/" + b;
+  if (pathBaseDir.empty()) {
+    pathBaseDir = std::filesystem::current_path();
   }
 
-  return a + "/" + b;
-}
-
-std::string getWorkingDirectory() {
-  char buffer[PATH_MAX];
-  return (getcwd(buffer, sizeof(buffer)) != nullptr ? std::string(buffer)
-                                                    : std::string(""));
+  return (pathBaseDir / pathRelativeFilePath).string();
```

Comment:
This method contains a bit of implicit magic, namely:

1. If the second argument is already absolute, then the first argument is completely ignored.
2. If the first argument is empty, then the current working directory is used instead.

While this works, of course, I think we could do away with a bit of this implicitness. I would instead assert that `baseDir` is non-empty. I'd document the behaviour that the second argument is chosen as-is if it is absolute.

Even better would be for this function to accept to paths rather than strings, in my opinion. Would that be possible.